experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -81,11 +89,41 @@ function cartan_matrix(R::RootSystem)
   return R.cartan_matrix
 end
 
-#function basis(R::RootSystem)
-#  rk = rank(R)
-#  it = 1:rk
-#  return [RootSpaceElem(R, matrix(QQ, 1, rk, i .== it)) for i in 1:rk]
-#end
+@doc raw"""
+    dual(R::RootSystem) -> RootSystem
+
+Returns the dual root system of `R`.
+"""
+@attr RootSystem function dual(R::RootSystem)
+  # TODO: find an algorithm that computes the dual reflection table directly
+  D = root_system(transpose(cartan_matrix(R)))
+  set_attribute!(D, :dual, R)
+  return D
+end
+
+@doc raw"""
+    coroot(R::RootSystem, i::Int) -> RootSpaceElem
+
+Returns the `i`-th coroot of `R`, i.e. the `i`-th root of the dual root system of `R`.
+This is a more efficient version for `coroots(R)[i]`.
+
+Also see: `coroots`.
```

Comment:
How about using an actual ref?
```suggestion
Also see: [`coroots`](@ref).
```

Reply:
This does only work once we have a markdown page collecting the docstrings. That's something @felix-roehrich wants to approach in the near future.